lib/osut/utils.rb

```diff
@@ -4214,32 +4327,43 @@ def boundedBox(pts = nil)
 
   ##
   # Generates re-'aligned' polygon vertices wrt main axis of symmetry of its
-  # largest bounded box. A Hash is returned with 6x key:value pairs ...
+  # largest bounded box. Input polygon vertex Z-axis values must equal 0, and be
+  # counterclockwise. A Hash is returned with 6x key:value pairs ...
   # set: realigned (cloned) polygon vertices, box: its bounded box (wrt to :set),
   # bbox: its bounding box, t: its translation transformation, r: its rotation
   # transformation, and o: the origin coordinates of its axis of rotation. First,
   # cloned polygon vertices are rotated so the longest axis of symmetry of its
   # bounded box lies parallel to the X-axis; :o being the midpoint of the narrow
-  # side (of the bounded box) nearest to grid origin (0,0,0). Once rotated,
+  # side (of the bounded box) nearest to grid origin (0,0,0). If the axis of
+  # symmetry of the boudned box is already parallel to the X-axis, then the
+  # rotation step is skipped (unless force == true). Whether rotated or not,
   # polygon vertices are then translated as to ensure one or more vertices are
   # aligned along the X-axis and one or more vertices are aligned along the
-  # Y-axis (no vertices with negative X or Y coordinate values). To unalign the
-  # returned set of vertices (or its bounded box, or its bounding box), first
-  # inverse the translation transformation, then inverse the rotation
+  # Y-axis (no vertices with negative X or Y coordinate values). To unalign
+  # the returned set of vertices (or its bounded box, or its bounding box),
+  # first inverse the translation transformation, then inverse the rotation
   # transformation.
   #
   # @param pts [Set<OpenStudio::Point3d>] OpenStudio 3D points
+  # @param force [Bool] whether to force rotation for aligned yet narrow boxes
   #
   # @return [Hash] :set, :box, :bbox, :t, :r & :o
   # @return [Hash] :set, :box, :bbox, :t, :r & :o (nil) if invalid (see logs)
-  def getRealignedFace(pts = nil)
+  def getRealignedFace(pts = nil, force = false)
     mth = "OSut::#{__callee__}"
     out = { set: nil, box: nil, bbox: nil, t: nil, r: nil, o: nil }
     pts = poly(pts, false, true)
     return out if pts.empty?
     return invalid("aligned plane", mth, 1, DBG, out) unless xyz?(pts, :z)
     return invalid("clockwise pts", mth, 1, DBG, out)     if clockwise?(pts)
 
+    # Optionally force rotation so bounded box ends up wider than taller.
+    # Strongly suggested for flat surfaces like roofs (see 'sloped?').
+    unless [true, false].include?(force)
+      log(DBG, "Ignoring force input (#{mth})")
+      force = false
+    end
```

Comment:
This option became key to systematize the treatment of roof surfaces, whether flat, low-sloped, etc.